ortho_seq_code/plotclass.py

```diff
@@ -0,0 +1,361 @@
+        omit_zeroes=True,
+        bin_color=None,
+        border=True,
+        out_dir=None,
+    ):
+        if alphabet_item is not None and site is not None:
+            print("Only one of site and alphabet_item may not be null.")
+            return
+        if alphabet_item is not None and alphabet_item not in self.alphbt_input:
+            print("Alphabet item must be one of:")
+            print(self.alphbt_input)
+            return
+        if site is not None:
+            if site > self.s or site < 0:
+                print("Site must be between 0 and", str(self.s))
+                return
+        if alphabet_item is not None:
+            a = self.alphbt_input.index(alphabet_item) * self.s
+            x_red = self.x_flat[a : a + self.d]
+        elif site is not None:
+            x_red = self.x[site]
+        else:
+            x_red = self.x_flat
+        if omit_zeroes:
+            x_red = np.array(x_red)[np.array(x_red) != 0].tolist()
+        if border:
+            if bins is not None:
+                width = 1 - bins / 200
+            else:
+                width = 1 - len(x_red) / 1200
+        else:
+            width = 0
+        plt.hist(x_red, bins=bins, color=bin_color, lw=width, ec="black")
+        plt.ylabel(self.phenotype)
+        if out_dir is not None:
+            path_sav = "rFon1D_hist_" + str(self.phenotype) or "" + ".png"
+            path_sav = path_sav.replace(" ", "_")
+            plt.savefig(os.path.join(str(out_dir), path_sav), dpi=400)
+            print(
+                "saved regression graph as", str(os.path.join(str(out_dir), path_sav)),
+            )
+        elif self.out_dir is not None:
+            path_sav = "rFon1D_hist_" + str(self.phenotype) or "" + ".png"
+            path_sav = path_sav.replace(" ", "_")
+            plt.savefig(os.path.join(str(self.out_dir), path_sav), dpi=400)
+            print(
+                "saved regression graph as",
+                str(os.path.join(str(self.out_dir), path_sav)),
+            )
+
+    def heatmap(self, out_dir=None):
+        fig, ax = plt.subplots(figsize=(8, 8))
+        im = ax.imshow(self.x)
+        ax.set_xticks(np.arange(len(self.alphbt_input)))
+        ax.set_xticklabels(self.alphbt_input)
+        ax.set_yticks(np.arange(self.s))
+        fig.tight_layout()
+        for i in range(self.s):
+            for j in range(self.d):
+                print(round(self.x[i, j], 2))
+                text = ax.text(
+                    j, i, round(self.x[i, j], 2), ha="center", va="center", color="w"
+                )
+        if out_dir is not None:
+            path_sav = "rFon1D_heatmap_" + str(self.phenotype) or "" + ".png"
+            path_sav = path_sav.replace(" ", "_")
+            plt.savefig(os.path.join(str(out_dir), path_sav), dpi=400)
+            print(
+                "saved regression graph as", str(os.path.join(str(out_dir), path_sav)),
+            )
+        elif self.out_dir is not None:
```

Comment:
this condition and above condition do the same thing can that if loop and this loop be combined with if out_dir is not None or self.out_dir is not None? also, what is the difference between these two out_dirs?

Reply:
out_dir is not required for an rf1d object, so you can still create graphs but not save them. If I specify an out_dir when creating the graphs, I don't need to specify it every time I run the function. If I want it saved somewhere else, I run the function with a new out_dir.